src/hotspot/share/gc/shared/referenceProcessor.cpp

```diff
@@ -1068,75 +1068,23 @@ void ReferenceProcessor::preclean_discovered_references(BoolObjectClosure* is_al
                                                         EnqueueDiscoveredFieldClosure* enqueue,
                                                         YieldClosure* yield,
                                                         GCTimer* gc_timer) {
-  // These lists can be handled here in any order and, indeed, concurrently.
+  Ticks preclean_start = Ticks::now();
 
-  // Soft references
-  {
-    GCTraceTime(Debug, gc, ref) tm("Preclean SoftReferences", gc_timer);
-    log_reflist("SoftRef before: ", _discoveredSoftRefs, _max_num_queues);
-    for (uint i = 0; i < _max_num_queues; i++) {
-      if (yield->should_return()) {
-        return;
-      }
-      if (preclean_discovered_reflist(_discoveredSoftRefs[i], is_alive,
-                                      enqueue, yield)) {
-        log_reflist("SoftRef abort: ", _discoveredSoftRefs, _max_num_queues);
-        return;
-      }
-    }
-    log_reflist("SoftRef after: ", _discoveredSoftRefs, _max_num_queues);
-  }
+  uint worker_id = WorkerThread::current()->id();
```

Comment:
This is the id of the worker *thread* (i.e. number), not the `worker_id` passed to the `work` method (although it often corresponds). The caller needs to pass that `worker_id` separately.